services/web_fetch_service.py

```
@@ -0,0 +1,301 @@
+            if p.scheme not in ("http", "https"):
+                return False
+            if not p.hostname or p.username or p.password:
+                return False
+            try:
+                ip = ipaddress.ip_address(p.hostname)
+                if not ip.is_global:
+                    return False
+            except ValueError:
+                # Hostname; DNS resolution checks can be added later if needed.
+                pass
+            return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)